### PR TITLE
Fixing translation API deprecated in Studio Pro 8.6

### PR DESCRIPTION
**Pull request by diego-antonelli**

### CODE REVIEW

File: src/Pagination/utils/ContainerUtils.ts [hunk 1]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
     targetListView.domNode.classList.remove("widget-pagination-loading");
 };
 
-export const mxTranslation = (namespace: string, key: string, replacements: any[]) => {
-    const templateString = mx.session.getConfig(`uiconfig.translations.${namespace}.${key}`)
-        || (window.mx.session.getConfig("uiconfig.translations") as any)[`${namespace}.${key}`]
-        || "[No translation]";
-    return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+export const mxTranslation = (namespace: string, key: string, replacements: any[], lookAtWindow: boolean) => {
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: Add fallback parameter  like `1 - 10 / 50`

File: src/Pagination/utils/ContainerUtils.ts [hunk 2]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+    } else {
+        return "[No translation]";
+    }
+};
+
+export const getTranslations = async (): Promise<void> => {
+    const localeCode = window.mx.session.getConfig("locale.code");
+    const metaData = await fetch("/metamodel.json");
+    const metadataJson = await metaData.json();
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: Double check IE11

File: src/Pagination/utils/ContainerUtils.ts [hunk 3]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
+    if (metadataJson && metadataJson.systemTexts) {
+        const systemTexts = metadataJson.systemTexts;
+        const localeIndex = metadataJson.languages.indexOf(localeCode);
+        window.translations = Object.keys(systemTexts).reduce((translations, currentKey) => ({ ...translations, [currentKey]: systemTexts[currentKey][localeIndex] }), {});
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: use  `window.__widgets_translations`

File: src/Pagination/utils/ContainerUtils.ts [hunk 4]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
+            || "[No translation]";
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+    } else {
+        return "[No translation]";
+    }
+};
+
+export const getTranslations = async (): Promise<void> => {
+    const localeCode = window.mx.session.getConfig("locale.code");
+    const metaData = await fetch("/metamodel.json");
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: make sure client does not die, only the widget if the URL is not available

File: src/Pagination/utils/ContainerUtils.ts [hunk 5]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
+    const metadataJson = await metaData.json();
+    if (metadataJson && metadataJson.systemTexts) {
+        const systemTexts = metadataJson.systemTexts;
+        const localeIndex = metadataJson.languages.indexOf(localeCode);
+        window.translations = Object.keys(systemTexts).reduce((translations, currentKey) => ({ ...translations, [currentKey]: systemTexts[currentKey][localeIndex] }), {});
+    } else {
+        logger.debug("Error while loading translations");
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: debug or error?

File: src/Pagination/components/PaginationContainer.ts [hunk 6]
```diff
@@ -65,12 +68,18 @@ class PaginationContainer extends Component<ModelerProps, PaginationContainerSta
...
             pageSize: this.viewStateManager.getPageState("pageSize", undefined), // We dont know, based on the modeler configuration of the list view.
             offset: this.viewStateManager.getPageState("offset", 0),
-            isPersisted: this.viewStateManager.getPageState("isPersisted", false)
+            isPersisted: this.viewStateManager.getPageState("isPersisted", false),
+            loadTranslations: false
         };
     }
 
-    componentDidMount() {
+    async componentDidMount() {
```
Comment (Andries-Smit) [inline src/Pagination/components/PaginationContainer.ts]: 👍 nice one!

File: src/Pagination/utils/ContainerUtils.ts (hunk 2 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Works fine

File: src/Pagination/utils/ContainerUtils.ts (hunk 4 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Fixed

File: src/Pagination/utils/ContainerUtils.ts (hunk 5 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Error, fixed

File: src/Pagination/utils/ContainerUtils.ts (hunk 3 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Fixed

File: src/Pagination/utils/ContainerUtils.ts (hunk 1 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Fixed

File: src/Pagination/utils/ContainerUtils.ts [hunk 7]
```diff
@@ -1,5 +1,7 @@
 import { DataSourceHelperListView } from "../../Shared/DataSourceHelper/DataSourceHelper";
 
+declare const window: any;
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: no.... thats a bit rough
``` ts
export interface CustomWindow extends Window {
    __widgets_translations: any;
}
declare let window: CustomWindow;

Or something like https://github.com/mendixlabs/charts/blob/f1e5b786182aeebc4ea36d95a377c0284154806b/src/components/PlotlyChart.ts#L29
```

File: src/Pagination/utils/ContainerUtils.ts [hunk 8]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
-    return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+export const mxTranslation = (namespace: string, key: string, replacements: any[], lookAtWindow: boolean, defaultTemplateValue: string) => {
+    if (!lookAtWindow) {
+        const templateString = mx.session.getConfig(`uiconfig.translations.${namespace}.${key}`)
+            || (window.mx.session.getConfig("uiconfig.translations") as any)[`${namespace}.${key}`]
+            || "[No translation]";
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+    } else if (window.__widgets_translations) {
+        const templateString = window.__widgets_translations[`${namespace}.${key}`]
+            || "[No translation]";
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]:
```suggestion
            || defaultTemplateValue || "[No translation]";
```

File: src/Pagination/utils/ContainerUtils.ts [hunk 9]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
-export const mxTranslation = (namespace: string, key: string, replacements: any[]) => {
-    const templateString = mx.session.getConfig(`uiconfig.translations.${namespace}.${key}`)
-        || (window.mx.session.getConfig("uiconfig.translations") as any)[`${namespace}.${key}`]
-        || "[No translation]";
-    return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+export const mxTranslation = (namespace: string, key: string, replacements: any[], lookAtWindow: boolean, defaultTemplateValue: string) => {
+    if (!lookAtWindow) {
+        const templateString = mx.session.getConfig(`uiconfig.translations.${namespace}.${key}`)
+            || (window.mx.session.getConfig("uiconfig.translations") as any)[`${namespace}.${key}`]
+            || "[No translation]";
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]:
```suggestion
            || defaultTemplateValue || "[No translation]";
```

File: src/Pagination/utils/ContainerUtils.ts [hunk 10]
```diff
@@ -49,9 +51,30 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
...
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+    } else if (window.__widgets_translations) {
+        const templateString = window.__widgets_translations[`${namespace}.${key}`]
+            || "[No translation]";
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
+    } else {
+        return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), defaultTemplateValue);
+    }
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]: In the if statements set the `templateString`, and a use single return.
``` ts
return replacements.reduce((substituteMessage, value, index) => substituteMessage.split("{" + (index + 1) + "}").join(value), templateString);
```

File: src/Pagination/utils/ContainerUtils.ts (hunk 9 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Done

File: src/Pagination/utils/ContainerUtils.ts (hunk 8 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Done

File: src/Pagination/utils/ContainerUtils.ts (hunk 7 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Done

File: src/Pagination/utils/ContainerUtils.ts (hunk 10 above)
Comment (diego-antonelli) [reply src/Pagination/utils/ContainerUtils.ts]: Done

File: src/Pagination/utils/ContainerUtils.ts [hunk 11]
```diff
@@ -49,9 +62,28 @@ export const hideLoader = (targetListView: DataSourceHelperListView) => {
     targetListView.domNode.classList.remove("widget-pagination-loading");
 };
 
-export const mxTranslation = (namespace: string, key: string, replacements: any[]) => {
-    const templateString = mx.session.getConfig(`uiconfig.translations.${namespace}.${key}`)
-        || (window.mx.session.getConfig("uiconfig.translations") as any)[`${namespace}.${key}`]
-        || "[No translation]";
+export const mxTranslation = (namespace: string, key: string, replacements: any[], lookAtWindow: boolean, defaultTemplateValue: string) => {
+    let templateString = "[No translation]";
```
Comment (Andries-Smit) [inline src/Pagination/utils/ContainerUtils.ts]:
```suggestion
    let templateString = defaultTemplateValue ?? "[No translation]";
```

**Pull request merged**